stdlib/asm/crypto/dsa/rpo_falcon512.masm

```diff
@@ -561,21 +620,28 @@ end
 #! Input: [s2_ptr, ...]
 #! Output: [norm_sq(s2), ...]
 #!
-#! Cycles: 13322
+#! Cycles: 11137
 export.compute_s2_norm_sq
     repeat.128
         padw
-        dup.4 add.1 swap.5
+        dup.4
         mem_loadw
 
-        repeat.4
-            exec.norm_sq
-            movdn.4
-        end
-
+        exec.norm_sq
+        swap
+        exec.norm_sq
+        add
+        swap
+        exec.norm_sq
+        add
+        swap
+        exec.norm_sq
+        add
+        swap
+        add.1
```

Comment:
Could you add comments that tracks what happens on the stack in this block?

Reply:
Sure, added comments to help track what's happening